From d16a3b8c69949125a839884f2fe79651ed876b8a Mon Sep 17 00:00:00 2001
From: ci7lus <7887955+ci7lus@users.noreply.github.com>
Date: Mon, 5 Jan 2026 22:30:36 +0900
Subject: [PATCH 2/2] avformat/demux: set unpacketized on mmttlv

---
 modules/demux/avformat/demux.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/modules/demux/avformat/demux.c b/modules/demux/avformat/demux.c
index f41169ce34..03a4658ae1 100644
--- a/modules/demux/avformat/demux.c
+++ b/modules/demux/avformat/demux.c
@@ -559,6 +559,7 @@ int avformat_OpenDemux( vlc_object_t *p_this )
                 es_fmt.video.i_sar_den = ar.den;
                 es_fmt.video.i_sar_num = ar.num;
             }
+
             break;

         case AVMEDIA_TYPE_SUBTITLE:
@@ -620,6 +621,8 @@ int avformat_OpenDemux( vlc_object_t *p_this )
             }
             break;
         }
+        if( !strcmp( p_sys->fmt->name, "mmttlv" ) )
+            es_fmt.b_packetized = false;

         AVDictionaryEntry *language = av_dict_get( s->metadata, "language", NULL, 0 );
         if ( language && language->value )
--
2.39.1

