From 7f62fd2edddf738c7d998a2a21f4e9273e5b686f Mon Sep 17 00:00:00 2001
From: ci7lus <7887955+ci7lus@users.noreply.github.com>
Date: Tue, 6 Jan 2026 18:13:44 +0900
Subject: [PATCH] codec/avcodef: Add TTML mapping to fourcc

---
 modules/codec/avcodec/fourcc.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/modules/codec/avcodec/fourcc.c b/modules/codec/avcodec/fourcc.c
index 394801f74c..90c20c2f91 100644
--- a/modules/codec/avcodec/fourcc.c
+++ b/modules/codec/avcodec/fourcc.c
@@ -523,6 +523,8 @@ static const struct vlc_avcodec_fourcc spu_codecs[] =
     { VLC_CODEC_BD_PG, AV_CODEC_ID_HDMV_PGS_SUBTITLE },
     { VLC_CODEC_BD_TEXT, AV_CODEC_ID_HDMV_TEXT_SUBTITLE },
     { VLC_CODEC_TELETEXT, AV_CODEC_ID_DVB_TELETEXT },
+    { VLC_CODEC_TTML, AV_CODEC_ID_TTML },
+    { VLC_CODEC_TTML_TS, AV_CODEC_ID_TTML },
     /* AV_CODEC_ID_SRT */
     /* ffmpeg only: AV_CODEC_ID_MICRODVD */
     /* ffmpeg only: AV_CODEC_ID_EIA_608 */
-- 
2.39.1

