From 703f49356b0d89f1a98671dcb087c4dd31f745bd Mon Sep 17 00:00:00 2001
From: ci7lus <7887955+ci7lus@users.noreply.github.com>
Date: Sat, 10 Jan 2026 02:35:22 +0900
Subject: [PATCH] demux/mpeg: skip SeekToTime always

---
 modules/demux/mpeg/ts.c | 59 ++++++++++++++---------------------------
 1 file changed, 20 insertions(+), 39 deletions(-)

diff --git a/modules/demux/mpeg/ts.c b/modules/demux/mpeg/ts.c
index a2001aca81..35c3e832dc 100644
--- a/modules/demux/mpeg/ts.c
+++ b/modules/demux/mpeg/ts.c
@@ -964,44 +964,14 @@ static int Control( demux_t *p_demux, int i_query, va_list args )
         break;

     case DEMUX_SET_POSITION:
+    {
         f = va_arg( args, double );
-        b_bool = (bool) va_arg( args, int ); /* precise */
+        int b_precise = va_arg( args, int );
+        VLC_UNUSED(b_precise);

         if(!p_sys->b_canseek)
             break;

-        if( p_sys->b_access_control &&
-           !p_sys->b_ignore_time_for_positions && b_bool && p_pmt )
-        {
-            time_t i_time, i_length = 0;
-            vlc_tick_t i_seektime = VLC_TICK_0 + vlc_tick_from_sec( i_length * f );
-            if( !EITCurrentEventTime( p_pmt, p_sys, &i_time, &i_length ) &&
-                 i_length > 0 && !SeekToTime( p_demux, p_pmt, i_seektime ) )
-            {
-                ReadyQueuesPostSeek( p_demux );
-                es_out_Control( p_demux->out, ES_OUT_SET_NEXT_DISPLAY_TIME, i_seektime );
-                return VLC_SUCCESS;
-            }
-        }
-
-        if( !p_sys->b_ignore_time_for_positions && b_bool && p_pmt &&
-             p_pmt->pcr.i_first != VLC_TICK_INVALID &&
-             p_pmt->i_last_dts != VLC_TICK_INVALID &&
-             p_pmt->pcr.i_current != VLC_TICK_INVALID )
-        {
-            vlc_tick_t i_length = p_pmt->i_last_dts - p_pmt->pcr.i_first;
-            i64 = p_pmt->pcr.i_first + i_length * f;
-            if( i64 <= p_pmt->i_last_dts )
-            {
-                if( !SeekToTime( p_demux, p_pmt, i64 ) )
-                {
-                    ReadyQueuesPostSeek( p_demux );
-                    es_out_Control( p_demux->out, ES_OUT_SET_NEXT_DISPLAY_TIME, i64 );
-                    return VLC_SUCCESS;
-                }
-            }
-        }
-
         if( vlc_stream_GetSize( p_sys->stream, &u64 ) == VLC_SUCCESS &&
             vlc_stream_Seek( p_sys->stream, (uint64_t)(u64 * f) ) == VLC_SUCCESS )
         {
@@ -1009,18 +979,29 @@ static int Control( demux_t *p_demux, int i_query, va_list args )
             return VLC_SUCCESS;
         }
         break;
+    }

     case DEMUX_SET_TIME:
     {
         vlc_tick_t i_time = va_arg( args, vlc_tick_t );

-        if( p_sys->b_canseek && p_pmt && p_pmt->pcr.i_first != VLC_TICK_INVALID &&
-           !SeekToTime( p_demux, p_pmt, p_pmt->pcr.i_first + i_time ) )
+        if( p_sys->b_canseek && p_pmt &&
+           ( p_pmt->pcr.i_first != VLC_TICK_INVALID || p_pmt->pcr.i_first_dts != VLC_TICK_INVALID ) &&
+             p_pmt->i_last_dts != VLC_TICK_INVALID )
         {
-            ReadyQueuesPostSeek( p_demux );
-            es_out_Control( p_demux->out, ES_OUT_SET_NEXT_DISPLAY_TIME,
-                            p_pmt->pcr.i_first + i_time - VLC_TICK_0 );
-            return VLC_SUCCESS;
+            vlc_tick_t i_start = (p_pmt->pcr.i_first != VLC_TICK_INVALID) ? p_pmt->pcr.i_first :
+                                  p_pmt->pcr.i_first_dts;
+            vlc_tick_t i_last = p_pmt->i_last_dts + p_pmt->pcr.i_pcroffset;
+            if( i_last > i_start )
+            {
+                double f = (double)i_time / (i_last - i_start);
+                if( vlc_stream_GetSize( p_sys->stream, &u64 ) == VLC_SUCCESS &&
+                    vlc_stream_Seek( p_sys->stream, (uint64_t)(u64 * f) ) == VLC_SUCCESS )
+                {
+                    ReadyQueuesPostSeek( p_demux );
+                    return VLC_SUCCESS;
+                }
+            }
         }
         break;
     }
--
2.39.1

