From 46c218143e74db375968cece0022a1497938e4f8 Mon Sep 17 00:00:00 2001
From: ci7lus <7887955+ci7lus@users.noreply.github.com>
Date: Mon, 2 Feb 2026 21:02:59 +0900
Subject: [PATCH] demux/mpeg: Remove fixup on DEMUX_GET_POSITION

---
 modules/demux/mpeg/ts.c | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/modules/demux/mpeg/ts.c b/modules/demux/mpeg/ts.c
index 55c47f3926..54445c8205 100644
--- a/modules/demux/mpeg/ts.c
+++ b/modules/demux/mpeg/ts.c
@@ -939,22 +939,6 @@ static int Control( demux_t *p_demux, int i_query, va_list args )
             }
         }
 
-        if( !p_sys->b_ignore_time_for_positions &&
-             p_pmt &&
-             p_pmt->pcr.i_first != VLC_TICK_INVALID &&
-             p_pmt->i_last_dts != VLC_TICK_INVALID &&
-             p_pmt->pcr.i_current != VLC_TICK_INVALID )
-        {
-            double i_length = p_pmt->i_last_dts - p_pmt->pcr.i_first;
-            i_length += p_pmt->pcr.i_pcroffset;
-            double i_pos = p_pmt->pcr.i_current - p_pmt->pcr.i_first;
-            if( i_length > 0 )
-            {
-                *pf = i_pos / i_length;
-                return VLC_SUCCESS;
-            }
-        }
-
         if( vlc_stream_GetSize( p_sys->stream, &u64 ) == VLC_SUCCESS )
         {
             uint64_t offset = vlc_stream_Tell( p_sys->stream );
-- 
2.39.1

