From be2fc1b3d74994eeb03d6c3f4f7598a917f7e184 Mon Sep 17 00:00:00 2001
From: ci7lus <7887955+ci7lus@users.noreply.github.com>
Date: Tue, 17 Feb 2026 01:28:50 +0900
Subject: [PATCH 9013/9014] demux/mpeg: Adjust ARIB-related network time to JST

---
 modules/demux/mpeg/ts_si.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/modules/demux/mpeg/ts_si.c b/modules/demux/mpeg/ts_si.c
index e2ab37d13a..a52efc85b6 100644
--- a/modules/demux/mpeg/ts_si.c
+++ b/modules/demux/mpeg/ts_si.c
@@ -402,7 +402,7 @@ static void TDTCallBack( void *opaque, dvbpsi_tot_t *p_tdt )
            but as there's no DST in JAPAN (since Showa 27/1952)
            and considering that no-one seems to send TDT or desc 0x58,
            falling back on fixed offset is safe */
-        p_sys->i_network_time += 9 * 3600;
+        p_sys->i_network_time -= 9 * 3600;
     }
 
     /* Because libdvbpsi is broken and deduplicating timestamp tables,
@@ -544,7 +544,7 @@ static void EITCallBack( void *opaque, dvbpsi_eit_t *p_eit )
         if( p_sys->standard == TS_STANDARD_ARIB )
         {
             /* See comments on TDT callback */
-            i_start += 9 * 3600;
+            i_start -= 9 * 3600;
         }
 
         msg_Dbg( p_demux, "  * event id=%"PRIu16" start_time:%"PRId64" duration=%d "
-- 
2.39.1

