From 3af8421a06571d3220d7adab9af3946b47f53cf7 Mon Sep 17 00:00:00 2001
From: ci7lus <7887955+ci7lus@users.noreply.github.com>
Date: Thu, 5 Mar 2026 01:39:29 +0900
Subject: [PATCH 1/2] fix: frame to bounds

---
 modules/video_output/window_macosx.m | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/modules/video_output/window_macosx.m b/modules/video_output/window_macosx.m
index ee9adbf7ab..b591325f4b 100644
--- a/modules/video_output/window_macosx.m
+++ b/modules/video_output/window_macosx.m
@@ -535,11 +535,11 @@ NS_ASSUME_NONNULL_BEGIN
 
 - (void)mouseMovedInternal:(NSEvent *)event
 {
-    NSPoint pointInView = 
+    NSPoint pointInView =
         [self convertPoint:event.locationInWindow fromView:nil];
     if ([self mouse:pointInView inRect:self.bounds]) {
         // Invert Y coordinates
-        CGPoint pointInWindow = 
+        CGPoint pointInWindow =
             CGPointMake(pointInView.x, self.bounds.size.height - pointInView.y);
         NSPoint pointInBacking = [self convertPointToBacking:pointInWindow];
         [_moduleDelegate reportMouseMoved:pointInBacking];
@@ -787,7 +787,7 @@ static int EmbedOpen(vlc_window_t *wnd)
         __block VLCVideoWindowContentView *contentView;
         dispatch_sync(dispatch_get_main_queue(), ^{
             contentView = [[VLCVideoWindowContentView alloc] initWithModuleDelegate:moduleDelegate];
-            contentView.frame = drawable.frame;
+            contentView.frame = drawable.bounds;
             [drawable addSubview:contentView];
         });
         if (unlikely(contentView == nil))
-- 
2.39.1

