From 1fa9fc04b3e751d4b5e876a84b3f0c213ffd2947 Mon Sep 17 00:00:00 2001
From: ci7lus <7887955+ci7lus@users.noreply.github.com>
Date: Mon, 4 May 2026 14:44:58 +0900
Subject: [PATCH] fix: ProbeDuration for changing PCR PID

---
 modules/demux/mpeg/ts_psi.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/modules/demux/mpeg/ts_psi.c b/modules/demux/mpeg/ts_psi.c
index f9c518fe83c..e35cd4cbbf0 100644
--- a/modules/demux/mpeg/ts_psi.c
+++ b/modules/demux/mpeg/ts_psi.c
@@ -2261,9 +2261,10 @@ static void PMTCallBack( void *data, dvbpsi_pmt_t *p_dvbpsipmt )
         if( !p_pmt->b_last_dts_probed )
             ProbeStart( p_demux, p_pmt->i_number );
         p_pmt->b_last_dts_probed = true;
+        vlc_tick_t i_start = (p_pmt->pcr.i_first != VLC_TICK_INVALID) ? p_pmt->pcr.i_first : p_pmt->pcr.i_first_dts;
         if( p_pmt->i_last_dts != VLC_TICK_INVALID &&
-            p_pmt->i_last_dts < p_pmt->pcr.i_first_dts )
-            p_pmt->i_last_dts = TimeStampWrapAround( p_pmt->pcr.i_first_dts, p_pmt->i_last_dts );
+            p_pmt->i_last_dts < i_start )
+            p_pmt->i_last_dts = TimeStampWrapAround( i_start, p_pmt->i_last_dts );
     }
 
     dvbpsi_pmt_delete( p_dvbpsipmt );
-- 
2.39.1

