From 23f3284b95b47ac595314da47100f1d270fa5801 Mon Sep 17 00:00:00 2001
From: ci7lus <7887955+ci7lus@users.noreply.github.com>
Date: Sat, 21 Feb 2026 16:41:54 +0900
Subject: [PATCH 2/3] fix: rollback4

---
 src/input/es_out.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/input/es_out.c b/src/input/es_out.c
index a4d730019b..e3d24ff2ef 100644
--- a/src/input/es_out.c
+++ b/src/input/es_out.c
@@ -2087,8 +2087,6 @@ static void EsOutProgramEpg(es_out_sys_t *p_sys, input_source_t *source,
         p_pgrm->p_meta = vlc_meta_New();
     if( p_pgrm->p_meta )
     {
-        vlc_meta_Set( p_pgrm->p_meta, vlc_meta_ESNowPlaying, psz_name );
-
         char psz_event_start[32];
         char psz_event_duration[32];
 
-- 
2.39.1

