From 493f62c1de17e1d09b10d2a1f30063215cad280d Mon Sep 17 00:00:00 2001
From: ci7lus <7887955+ci7lus@users.noreply.github.com>
Date: Sat, 21 Feb 2026 16:42:18 +0900
Subject: [PATCH 3/3] fix: rollback5

---
 src/input/es_out.c | 16 ----------------
 1 file changed, 16 deletions(-)

diff --git a/src/input/es_out.c b/src/input/es_out.c
index e3d24ff2ef..099ffdafea 100644
--- a/src/input/es_out.c
+++ b/src/input/es_out.c
@@ -2163,23 +2163,7 @@ static void EsOutProgramEpg(es_out_sys_t *p_sys, input_source_t *source,
                 free( ppsz_extra_names );
             }
         }
-        const char *psz_nowplaying = p_pgrm->p_meta ?
-                                     vlc_meta_Get( p_pgrm->p_meta, vlc_meta_ESNowPlaying ) : NULL;
-
-        input_item_SetESNowPlaying( input_priv(p_input)->p_item, psz_nowplaying );
         input_SendEventMeta( p_input );
-
-        const char *now_playing_tr =
-            vlc_meta_TypeToLocalizedString(vlc_meta_ESNowPlaying);
-        int ret;
-        if( psz_nowplaying )
-            ret = input_item_AddInfo( p_item, psz_cat, now_playing_tr,
-                                      "%s", psz_nowplaying );
-        else
-            ret = input_item_DelInfo( p_item, psz_cat, now_playing_tr );
-
-        if( ret == VLC_SUCCESS )
-            input_SendEventMetaInfo( p_input );
     }
 
     free( psz_cat );
-- 
2.39.1

